libkbfs/prefetcher.go

```diff
@@ -496,6 +586,19 @@ func (p *blockPrefetcher) run(testSyncCh <-chan struct{}) {
 			p.log.Debug("canceling prefetch for block %s", blockID)
 			// Walk up the block tree and delete every parent.
 			p.applyToParentsRecursive(p.cancelPrefetch, blockID, pre)
+		case reqInt := <-p.prefetchRescheduleCh.Out():
+			req := reqInt.(*prefetchRequest)
+			blockID := req.ptr.ID
+			pre, ok := p.prefetches[blockID]
+			if !ok {
+				pre = p.newPrefetch(1, false, req)
+				p.prefetches[blockID] = pre
+			}
```

Comment:
Doing the following will allow you to avoid passing `req` into the `rescheduleTopBlock` method.
```go
} else if (pre.req == nil) {
	pre.req = req
}
```